src/dash-playlist-loader.js

```diff
@@ -336,35 +358,37 @@ export default class DashPlaylistLoader extends EventTarget {
     });
   }
 
+  haveMaster_() {
+    this.state = 'HAVE_MASTER';
+
+    if (!this.masterPlaylistLoader_) {
+      this.master = this.parseMasterXml();
+      // We have the master playlist at this point, so
+      // trigger this to allow MasterPlaylistController
+      // to make an initial playlist selection
+      this.trigger('loadedplaylist');
+
+    } else if (!this.media_) {
```

Comment:
I see where you are coming from 🤔 . Here are the reasons I eventually decided on this path:

i. it did not appear to actually be used by the main loader of either the PlaylistLoader or DashPlaylistLoader in playback 

ii. it was being used only in the tests which made it seem like this was a common case and resulted in a lot of tests being written assuming that an initial (video) playlist would be set.

iii. `!this.media_` isn't the best equivalent of whether there is a pending request to set media since `this.media_` indicates that the media has already been set. However, this will likely change once we include support for `sidx`, which we could then use `!this.request && !this.media_` to check for which I think would cover both cases?

Reply:
My main concern is that this relies on the behavior that MPC will be making an initial selection after `loadedplaylist` instead of being more independent in that it will trigger the event so something can make an initial selection, but has a fallback. So any change to MPC behavior means the dev also needs to know that a change might need to be made here. It also means we have to have a lot more boilerplate in the the tests to set an initial media when we are trying to test some other behavior.

wrt iii. I agree checking `meia_` isn't really a good equivalent here. I think just checking `request_` would work just like it does in HLS. This is another reason to maintain the asynchronous nature of `media()` even when there is no sidx so it could be synchronous. By keeping async with a setTimeout, you can set `request_`  to that when no sidx is present so behavior aligns with the PlaylistLoader. Basically your `request_` is either for a sidx, or just an asnyc fn call added to the queue

Reply:
>It also means we have to have a lot more boilerplate in the the tests to set an initial media when we are trying to test some other behavior.

I actually think this is a good thing as it's clear what is actually happening in the test rather than having to reference the source code to see that x, y, z methods are called automatically. 

I'll go down the path of setting `request_` to the asynchronously called methods too

Reply:
One thing to note is that `this.master` doesn't exist for the child loader, right? So this would be the fallback for the master loader only (I hope)

Reply:
correct, the child loader shouldn't have a `this.master`